### PR TITLE
feat: hideRepliesToBlockedMessages

### PR DESCRIPTION
**Pull request by CodeF53**
Adds new option to NoBlockedMessages, default off:
Discord_bIarBcg95U

Without on
ShareX_rhH4I0OWJQ

With on
ShareX_wKHcRm2lWx

### CODE REVIEW

File: src/plugins/noBlockedMessages/index.ts [hunk 1]
```diff
@@ -60,9 +68,26 @@ export default definePlugin({
...
+    },
+
+    isReplyToBlocked(message: Message) {
+        if (!Settings.plugins.NoBlockedMessages.hideRepliesToBlockedMessages)
+            return false;
+
+        const { messageReference } = message;
+        if (!messageReference) return false;
+        const replyMessage = MessageStore.getMessage(messageReference.channel_id, messageReference.message_id);
+        return this.isBlocked(replyMessage);
```
Comment (Vendicated) [inline src/plugins/noBlockedMessages/index.ts:86]:
```suggestion
        if (!Settings.plugins.NoBlockedMessages.hideRepliesToBlockedMessages || !message)
            return false;

		try {
        	const { messageReference } = message;
	        if (!messageReference) return false;
	        
	        const replyMessage = MessageStore.getMessage(messageReference.channel_id, messageReference.message_id);
	        return this.isBlocked(replyMessage);
	    } catch (e) {
            new Logger("NoBlockedMessages").error("Failed to check if user is blocked:", e);
		}
```

File: src/plugins/noBlockedMessages/index.ts [hunk 2]
```diff
@@ -48,10 +49,17 @@ export default definePlugin({
...
-                    replace: (_, props) => `if($self.isBlocked(${props}.message))return;`
+                    replace: (_, props) => `if($self.isBlocked(${props}.message)||$self.isReplyToBlocked(${props}.message))return;`
                 }
             ]
-        }))
+        })),
+        {
+            find: ".messageListItem",
+            replacement: {
+                match: /(?<=\i=)(?=\(0,(\i)\.jsx)/,
```
Comment (Vendicated) [inline src/plugins/noBlockedMessages/index.ts:59]: way too generic, please make this more specific, by adding some kind of anchor

**Comment by Oroborius** [comment]
If possible add option to hide the user from userlist as well?

**Comment by Feathers8** [comment]
There is a different PR for that: #2170

**Pull request closed without merging**